pkg/extractor/extractor.go

```diff
@@ -69,33 +71,49 @@ type FileConfigurations struct {
 
 func ParseYaml(content string) (*[]Configuration, error) {
 	configurations, err := extractYamlConfigurations(content)
-	if err != nil {
+	if err != nil && err != io.EOF {
 		return nil, err
 	} else {
 		return configurations, nil
 	}
 }
 
+func ParseJsonToK8sValues(content []byte) (map[string]interface{}, error) {
+	var values map[string]interface{}
+	err := json.Unmarshal(content, &values)
+	if err != nil {
+		return nil, err
+	}
+	return values, nil
+}
+
 func extractYamlConfigurations(content string) (*[]Configuration, error) {
 	var configurations []Configuration
 
 	yamlDecoder := yaml.NewDecoder(bytes.NewReader([]byte(content)))
 
 	var err error
 	for {
-		var doc = map[string]interface{}{}
-		err = yamlDecoder.Decode(&doc)
+		var anyTypeStore interface{}
+		err = yamlDecoder.Decode(&anyTypeStore)
 		if err != nil {
-			break
+			if err == io.EOF {
+				break
+			} else {
+				return nil, err
+			}
 		}
 
-		if len(doc) > 0 {
-			configurations = append(configurations, doc)
+		yamlByteArray, err := yaml.Marshal(anyTypeStore)
```

Comment:
I miss your comments, it made it much more clear to understand why we decode every YAML, marshal it and then parse it to JSON.